src/filesystem.cpp

```diff
@@ -946,37 +1024,46 @@ LocalPath LocalPath::subpathFrom(size_t bytePos) const
     return LocalPath::fromLocalname(localpath.substr(bytePos));
 }
 
-
 void LocalPath::ensureWinExtendedPathLenPrefix()
 {
 #if defined(_WIN32) && !defined(WINDOWS_PHONE)
-    localpath.append("", 1);
-    if (!PathIsRelativeW((LPWSTR)localpath.c_str()) && ((localpath.size() < 4) || memcmp(localpath.data(), L"\\\\", 4)))
-        localpath.insert(0, (const char*)(const wchar_t*)L"\\\\?\\", 8);
-    localpath.resize(localpath.size() - 1);
+    if (!PathIsRelativeW(localpath.c_str()) && ((localpath.size() < 2) || memcmp(localpath.data(), L"\\\\", 4)))
+        localpath.insert(0, L"\\\\?\\", 4);
 #endif
 }
 
-string LocalPath::substrTo(size_t bytePos) const
+LocalPath LocalPath::subpathTo(size_t bytePos) const
 {
-    return localpath.substr(0, bytePos);
+    LocalPath p;
+    p.localpath = localpath.substr(0, bytePos);
+    return p;
 }
 
 string LocalPath::toPath(const FileSystemAccess& fsaccess) const
 {
     string path;
+#if defined(_WIN32)
+    fsaccess.local2path(const_cast<std::wstring*>(&localpath), &path); // todo: const correctness for local2path etc
+#else
     fsaccess.local2path(const_cast<string*>(&localpath), &path); // todo: const correctness for local2path etc
+#endif   
     return path;
 }
 
 string LocalPath::toName(const FileSystemAccess& fsaccess, FileSystemType fsType) const
 {
+#if defined(_WIN32)
+    std::wstring name = localpath;
```

Comment:
line# 1057 calls `FileSystemAccess::local2name(std::wstring* filename, FileSystemType fsType)` which changes `filename`. We don't intend to change original `localpath`

Reply:
> `FileSystemAccess::local2name(std::wstring* filename, FileSystemType fsType)` which changes `filename`.

IIRC, that is the newly introduced form and it _doesn't_ modify `filename`. Could you please check again? That's why changing it to receive `wstring const&` rather than `wstring*` would be useful.

Reply:
That's right. Made the required changes.